testing/dart/test_util.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.6
+
+import 'package:test/test.dart';
+
+/// Asserts that `callback` throws an [AssertionError].
+///
+/// When running in a VM in which assertions are enabled, asserts that the
+/// specified callback throws an [AssertionError]. When asserts are not
+/// enabled, such as when running using a release-mode VM with default
+/// settings, this acts as a no-op.
+void expectAssertion(Function callback) {
```

Comment:
random unsolicited note (since you asked :D) why not give it a class and a function to just expectThrows instead? Something like https://www.codota.com/code/java/methods/org.testng.Assert/expectThrows

Reply:
Interesting. Will land this as-is so I can use it for the lerp tests, but as I say, planning to do a cleanup so will take a look!